auparse/auparse.go

```diff
@@ -290,222 +308,224 @@ func normalizeAuditMessage(typ AuditMessageType, msg string) (string, error) {
 	return msg, nil
 }
 
-func extractKeyValuePairs(msg string, data map[string]*field) {
+func extractKeyValuePairs(msg string) fieldMap {
+	data := make(fieldMap, 0)
 	matches := kvRegex.FindAllStringSubmatch(msg, -1)
 	for _, m := range matches {
-		key := m[1]
-		f := newField(m[2])
-		f.Set(trimQuotesAndSpace(m[2]))
+		key, orig := m[1], m[2]
+		value := trimQuotesAndSpace(orig)
 
 		// Drop fields with useless values.
-		switch f.Value() {
+		switch value {
 		case "", "?", "?,", "(null)":
 			continue
 		}
 
 		if key == "msg" {
-			extractKeyValuePairs(f.Value(), data)
-		} else {
-			data[key] = f
+			for mk, mv := range extractKeyValuePairs(value) {
+				data.add(mk, mv)
+			}
+			continue
 		}
+		data.add(key, field{orig: orig, value: value})
 	}
+	return data
 }
 
 func trimQuotesAndSpace(v string) string { return strings.Trim(v, `'" `) }
 
 // Enrichment after KV parsing
-
-//nolint:errcheck // Continue enriching even if some fields do not exist.
```

Comment:
I prefer the linter comment to blank identifier assignment; it makes it clear why the errors are being ignored.

Reply:
Do you mean keeping 
```
// Continue enriching even if some fields do not exist.
```
as a separate comment line?
The lint exclusion `//nolint:errcheck` is not necessary anymore.